netort/resource.py

```diff
@@ -403,4 +418,135 @@ def readline(self):
             return ''
 
 
+class S3Opener(object):
+    """ Simple Storage Service opener
+        Downloads files.
+
+        s3credentials.json fmt:
+        {
+            "aws_access_key_id": "key-id",
+            "aws_secret_access_key": "secret-id",
+            "host": "hostname.tld",
+            "port": 7480,
+            "is_secure": false
+        }
+    """
+
+    def __init__(self, uri, credentials_path='/etc/yandex-tank/s3credentials.json'):
+        # read s3 credentials
+        # FIXME move to default config? which section and how securely store the keys?
```

Comment:
No, it's ok to store the credentials in external file.

Reply:
So, where do you suggest to store auth sensitive information? Config definitely is not the right place. And, overload plugin stores token just like that.

Reply:
I suggest external file =)

Reply:
troll.jpg